src/ui/overlay.rs

```diff
@@ -202,6 +206,32 @@ impl EguiOverlay for OverlayApp {
             self.last_paint_time = Instant::now();
             egui_context.request_repaint();
         }
+
+        // Update play-time tracker display
+        {
```

Comment:
Hmm, I see what you were going for here, but we're already updating the overlay every half-second when recording is active and on state transitions. I'd keep it simple: in the code below, render the text if required without any additional caching.

As part of that, I'd also directly use `format!()` with the template instead of using `.replace`. `PLAY_TIME_MESSAGE` isn't used anywhere else, and it's a display-only concern.